engines/mbic3.cpp

```diff
@@ -753,24 +781,33 @@ Conjunction ModelBasedIC3::generalize_predecessor(size_t i,
   return res;
 }
 
-bool ModelBasedIC3::intersects(const Term & A, const Term & B) const
+bool ModelBasedIC3::intersects(const Term & A, const Term & B)
 {
-  solver_->push();
+  push_solver_context();
   solver_->assert_formula(A);
   solver_->assert_formula(B);
   Result r = solver_->check_sat();
-  solver_->pop();
+  pop_solver_context();
   return r.is_sat();
 }
 
-bool ModelBasedIC3::intersects_initial(const Term & t) const
+bool ModelBasedIC3::intersects_initial(const Term & t)
 {
-  return intersects(ts_.init(), t);
+  return intersects(init_label_, t);
 }
 
 void ModelBasedIC3::assert_frame(size_t i) const
 {
-  solver_->assert_formula(get_frame(i));
+  assert(frame_labels_.size() == frames_.size());
+  Term assump;
+  for (size_t j = 0; j < frame_labels_.size(); ++j) {
+    assump = frame_labels_[j];
+    if (j < i) {
+      assump = solver_->make_term(Not, assump);
```

Comment:
this is an optimization, not a requirement. Right?

Reply:
Yep, that's right! I'll add a comment -- good catch.